### PR TITLE
CLI Global Options - Part 4 - Refactor main

### PR DESCRIPTION
**Pull request by tonylee80**
depends on #2207

Refactor `main` function to remove medplumClient as a param.

### Summary
🗑️🔧🚀

Removed `MedplumClient` dependency from CLI module and tests. Simplified the CLI `main` function signature and usage. Refactored the CLI tests to remove unnecessary imports, variables, and arguments.

> _`medplum` argument_
> _gone from CLI and tests_
> _autumn of cleanup_

### Walkthrough
*  Removed `MedplumClient` dependency from CLI commands and tests ([link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-32f0901bf847ac4f6b89540d53766e03661dce5f12d13a51d543452c15952643L45-L51), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-32f0901bf847ac4f6b89540d53766e03661dce5f12d13a51d543452c15952643L64-R57), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-32f0901bf847ac4f6b89540d53766e03661dce5f12d13a51d543452c15952643L106-R99), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-d305bdcb8aa2ed2bfc732d8af308cc2614a71a744b30158472e685f260669339L7-R9), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-d305bdcb8aa2ed2bfc732d8af308cc2614a71a744b30158472e685f260669339L95-R92), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-d305bdcb8aa2ed2bfc732d8af308cc2614a71a744b30158472e685f260669339L101-R98), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-fa825c9e9d946ea06b175f39b698f55ed5d7850bf2d02eadce3ad9885c5f495aL8), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-fa825c9e9d946ea06b175f39b698f55ed5d7850bf2d02eadce3ad9885c5f495aL17-L18), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-fa825c9e9d946ea06b175f39b698f55ed5d7850bf2d02eadce3ad9885c5f495aL90-R87), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-fa825c9e9d946ea06b175f39b698f55ed5d7850bf2d02eadce3ad9885c5f495aL153-R150), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-fa825c9e9d946ea06b175f39b698f55ed5d7850bf2d02eadce3ad9885c5f495aL223-R220), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-fa825c9e9d946ea06b175f39b698f55ed5d7850bf2d02eadce3ad9885c5f495aL285-R282), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-fa825c9e9d946ea06b175f39b698f55ed5d7850bf2d02eadce3ad9885c5f495aL343-R340), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-fa825c9e9d946ea06b175f39b698f55ed5d7850bf2d02eadce3ad9885c5f495aL404-R401), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-fa825c9e9d946ea06b175f39b698f55ed5d7850bf2d02eadce3ad9885c5f495aL465-R462), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-fedb3c5d5f0515d7f1673f50ae55d763a0f99b19d33ae819a45d7c57241db3c7L7-R9), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-fedb3c5d5f0515d7f1673f50ae55d763a0f99b19d33ae819a45d7c57241db3c7L113-R110), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-7c11a1043eb043e7ac74a0273db0f5e59537ee396a277735c96b4a9b649e1fe4L9), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-7c11a1043eb043e7ac74a0273db0f5e59537ee396a277735c96b4a9b649e1fe4L46-L47), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-7c11a1043eb043e7ac74a0273db0f5e59537ee396a277735c96b4a9b649e1fe4L196-R193), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-7c11a1043eb043e7ac74a0273db0f5e59537ee396a277735c96b4a9b649e1fe4L255-R252), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-7c11a1043eb043e7ac74a0273db0f5e59537ee396a277735c96b4a9b649e1fe4L264-R261), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-7c11a1043eb043e7ac74a0273db0f5e59537ee396a277735c96b4a9b649e1fe4L274-R271), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-7c11a1043eb043e7ac74a0273db0f5e59537ee396a277735c96b4a9b649e1fe4L284-R281), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-44250091a53bdf47291bad9e5a747a947ad04171e15aeecf48d07a9db522ec69L8-R10), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-44250091a53bdf47291bad9e5a747a947ad04171e15aeecf48d07a9db522ec69L139-R136), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-44250091a53bdf47291bad9e5a747a947ad04171e15aeecf48d07a9db522ec69L145-R142), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-44250091a53bdf47291bad9e5a747a947ad04171e15aeecf48d07a9db522ec69L151-R148), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-44250091a53bdf47291bad9e5a747a947ad04171e15aeecf48d07a9db522ec69L157-R154), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-ccbabf42b3b72371708d42e69a88eb8517ed9e7087b910bf4e51220e0f64956bL43-R43), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-ccbabf42b3b72371708d42e69a88eb8517ed9e7087b910bf4e51220e0f64956bL64-R64), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-ccbabf42b3b72371708d42e69a88eb8517ed9e7087b910bf4e51220e0f64956bL86-R86), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-ccbabf42b3b72371708d42e69a88eb8517ed9e7087b910bf4e51220e0f64956bL110-R110), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-ccbabf42b3b72371708d42e69a88eb8517ed9e7087b910bf4e51220e0f64956bL137-R137), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-ccbabf42b3b72371708d42e69a88eb8517ed9e7087b910bf4e51220e0f64956bL170-R170), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-ccbabf42b3b72371708d42e69a88eb8517ed9e7087b910bf4e51220e0f64956bL201-R201), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-ccbabf42b3b72371708d42e69a88eb8517ed9e7087b910bf4e51220e0f64956bL210-R220), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-ccbabf42b3b72371708d42e69a88eb8517ed9e7087b910bf4e51220e0f64956bL236-R227), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-ccbabf42b3b72371708d42e69a88eb8517ed9e7087b910bf4e51220e0f64956bL258-R249), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-ccbabf42b3b72371708d42e69a88eb8517ed9e7087b910bf4e51220e0f64956bL279-R270), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-ccbabf42b3b72371708d42e69a88eb8517ed9e7087b910bf4e51220e0f64956bL303-R294), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-ccbabf42b3b72371708d42e69a88eb8517ed9e7087b910bf4e51220e0f64956bL330-R321), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-ccbabf42b3b72371708d42e69a88eb8517ed9e7087b910bf4e51220e0f64956bL363-R354), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-ccbabf42b3b72371708d42e69a88eb8517ed9e7087b910bf4e51220e0f64956bL394-R385), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-ccbabf42b3b72371708d42e69a88eb8517ed9e7087b910bf4e51220e0f64956bL404-R395), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-ccbabf42b3b72371708d42e69a88eb8517ed9e7087b910bf4e51220e0f64956bL410-R406), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-ba0eec9b49430d3db287c8fc6a316004d00df70f9396382fef7d8a1c513824eeL146-R146), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-ba0eec9b49430d3db287c8fc6a316004d00df70f9396382fef7d8a1c513824eeL192-R192), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-ba0eec9b49430d3db287c8fc6a316004d00df70f9396382fef7d8a1c513824eeL216-R216), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-2538b20a8c1ae5193b3c362c6036a0394462cecb4cd5ba3f5d95d5159fd13af8L1-L2), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-2538b20a8c1ae5193b3c362c6036a0394462cecb4cd5ba3f5d95d5159fd13af8L18-L19), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-2538b20a8c1ae5193b3c362c6036a0394462cecb4cd5ba3f5d95d5159fd13af8L27), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-2538b20a8c1ae5193b3c362c6036a0394462cecb4cd5ba3f5d95d5159fd13af8L50-R45), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-2538b20a8c1ae5193b3c362c6036a0394462cecb4cd5ba3f5d95d5159fd13af8L58-R53), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-2fb448d11702324cc59d657d5945d59cd325540eb87b71f2e66a5bc01ada834cL1-R1), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-2fb448d11702324cc59d657d5945d59cd325540eb87b71f2e66a5bc01ada834cL10-R12), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-2fb448d11702324cc59d657d5945d59cd325540eb87b71f2e66a5bc01ada834cL64-R52), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-a74aca43c19d751210fc06813e841d7723f7e7ba31d1b62ed0b4ef1d68878384L59-R59), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-a74aca43c19d751210fc06813e841d7723f7e7ba31d1b62ed0b4ef1d68878384L81-R81), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-a74aca43c19d751210fc06813e841d7723f7e7ba31d1b62ed0b4ef1d68878384L117-R117), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-a74aca43c19d751210fc06813e841d7723f7e7ba31d1b62ed0b4ef1d68878384L153-R153), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-a74aca43c19d751210fc06813e841d7723f7e7ba31d1b62ed0b4ef1d68878384L159-R159), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-a74aca43c19d751210fc06813e841d7723f7e7ba31d1b62ed0b4ef1d68878384L176-R176), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-a74aca43c19d751210fc06813e841d7723f7e7ba31d1b62ed0b4ef1d68878384L201-R201), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-a74aca43c19d751210fc06813e841d7723f7e7ba31d1b62ed0b4ef1d68878384L236-R236), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-140694d422bd950999bf5ac1fa9df57b87fda8f3ed8687b75700f88513949986L33-R33), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-140694d422bd950999bf5ac1fa9df57b87fda8f3ed8687b75700f88513949986L45-R45), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-140694d422bd950999bf5ac1fa9df57b87fda8f3ed8687b75700f88513949986L51-R56), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-140694d422bd950999bf5ac1fa9df57b87fda8f3ed8687b75700f88513949986L62-R62), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-140694d422bd950999bf5ac1fa9df57b87fda8f3ed8687b75700f88513949986L69-R74), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-140694d422bd950999bf5ac1fa9df57b87fda8f3ed8687b75700f88513949986L80-R80), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-140694d422bd950999bf5ac1fa9df57b87fda8f3ed8687b75700f88513949986L92-R86))
* Simplified `main` function signature and calls in `packages/cli/src/index.ts` and tests ([link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-2fb448d11702324cc59d657d5945d59cd325540eb87b71f2e66a5bc01ada834cL64-R52), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-2538b20a8c1ae5193b3c362c6036a0394462cecb4cd5ba3f5d95d5159fd13af8L50-R45), [link](https://github.com/medplum/medplum/pull/2208/files?diff=unified&w=0#diff-2538b20a8c1ae5193b3c362c6036a0394462cecb4cd5ba3f5d95d5159fd13af8L58-R53))

### CODE REVIEW

File: packages/cli/src/index.ts [hunk 1]
```diff
@@ -61,24 +49,7 @@ export async function main(medplumClient: MedplumClient, argv: string[]): Promis
...
-    tokenUrl,
-    fhirUrlPath,
-    storage: new FileSystemStorage(),
-    onUnauthenticated: onUnauthenticated,
-  });
-
-  if (accessToken) {
-    medplumClient.setAccessToken(accessToken);
-  }
-  await main(medplumClient, process.argv);
```
Comment (tonylee80) [inline packages/cli/src/index.ts:81]: This logic has moved to https://github.com/medplum/medplum/pull/2197/files#diff-80e8d5a02c540100cc38f9c1240d73742010ee4b2f195e615943ac89d67d5bc7

File: packages/cli/src/index.ts [hunk 2]
```diff
@@ -7,21 +7,9 @@ import { bot, createBotDeprecate, deployBotDeprecate, saveBotDeprecate } from '.
...
-  medplum = medplumClient;
 
+export async function main(argv: string[]): Promise<void> {
   try {
-    const clientId = process.env['MEDPLUM_CLIENT_ID'];
-    const clientSecret = process.env['MEDPLUM_CLIENT_SECRET'];
-    if (clientId && clientSecret) {
-      medplumClient.setBasicAuth(clientId, clientSecret);
-      await medplumClient.startClientLogin(clientId, clientSecret);
-    }
```
Comment (tonylee80) [inline packages/cli/src/index.ts:24]: move to https://github.com/medplum/medplum/pull/2197/files#diff-80e8d5a02c540100cc38f9c1240d73742010ee4b2f195e615943ac89d67d5bc7

File: packages/cli/src/util/command.ts [hunk 3]
```diff
@@ -3,18 +3,23 @@ import { Command } from 'commander';
...
-
+    addGlobalOptions(cmd);
     return cmd;
   }
 }
 
 export function createMedplumCommand(name: string): MedplumCommand {
-  return new MedplumCommand(name);
+  const cmd = new MedplumCommand(name);
+  addGlobalOptions(cmd);
```
Comment (tonylee80) [inline packages/cli/src/util/command.ts:13]: This accounts for other commands that are not subcommands such as commands REST commands `get`, `post`, `patch`, `delete`, etc..

**Review by codyebberson — APPROVED** [review]
Nice.  Methodical.

**Pull request merged**